File: Java/1431. Kids With the Greatest Number of Candies.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int max = Arrays.stream(candies).max().getAsInt();
        List<Boolean> result = new ArrayList<>();

        for (int candy : candies)
            result.add(candy + extraCandies >= max);

        return result;
    }

    public List<Boolean> kidsWithCandiesStreamAPI(int[] candies, int extraCandies) {
        int max = Arrays.stream(candies).max().getAsInt();

        return Arrays.stream(candies).mapToObj(c -> c + extraCandies >= max).collect(Collectors.toList());
    }
}

File: TypeScript/1431. Kids With the Greatest Number of Candies.ts
function KidsWithCandies(candies: number[], extraCandies: number): boolean[] {
    const max = Math.max(...candies);
    return candies.map(c => c + extraCandies >= max);
};

function KidsWithCandiesLoop(candies: number[], extraCandies: number): boolean[] {
    const max = Math.max(...candies);
    const result: boolean[] = [];

    for (let i = 0; i < candies.length; i++) {
        if (candies[i] + extraCandies >= max)
            result.push(true);
        else
            result.push(false);

    }

    return result;
};